torch/_dynamo/output_graph.py

```diff
@@ -313,6 +320,12 @@ def restore_graphstate(self, state: OutputGraphState):
                 removed_nodes += 1
         log.debug(f"restore_graphstate: removed {removed_nodes} nodes")
 
+    def add_grapharg(self, arg: GraphArg):
+        curr_pos = len(self.graphargs)
+        self.graphargs.append(arg)
+        if isinstance(arg.source, LocalInputSource):
+            self.pos_to_arg[arg.source.pos] = curr_pos
```

Comment:
this is more me confirming my understanding than anything, but:

The reason we need this `pos_to_arg` dict looks like it's because:
(1) `GraphArg.source.pos` represents which python local you came from, as indexed by `co_varnames`
(2) `curr_pos` represents which GraphArg you are in dynamo's list of GraphArgs

Dynamo constructs its guard w.r.t. graph args (to print nice guard info, dynamo looks up source info about the i'th graph arg), so (2) is the important thing.

If I'm reading correctly, is (1) strictly necessary? (Could the GraphArg just directly store "I am the k'th graph arg" on itself directly, and have that be the info that we stash on the FakeTensor, that AOTAutograd picks up?) Maybe the reason for the indirection is that it's cleaner for GraphArg to only have to reason about things w.r.t its source. Just checking

Reply:
That's possible. I had a variation of this, but it required every call into the GraphArg ctor to do `len(self.graphargs)` for the sake of saving a single map lookup. What I was worried about was future instantiations of GraphArg forgetting to add that boilerplate. No strong opinion tho.